src/test/java/com/commercetools/sync/customobjects/utils/CustomObjectSyncUtilsTest.java

```diff
@@ -0,0 +1,249 @@
+
+        JsonNode oldNestedJson = JsonNodeFactory.instance.objectNode()
+                .put("username", "Peter")
+                .put("userId", "123-456-789");
+        JsonNode newNestedJson = JsonNodeFactory.instance.objectNode()
+                .put("userId", "123-456-789")
+                .put("username", "Peter");
+
+        JsonNode oldJsonNode = JsonNodeFactory.instance.objectNode()
+                .set("nestedJson", oldNestedJson);
+
+        JsonNode newJsonNode = JsonNodeFactory.instance.objectNode()
+                .set("nestedJson", newNestedJson);
+
+
+        newCustomObjectdraft = CustomObjectDraft.ofUnversionedUpsert(container, key, newJsonNode);
+        oldCustomObject = mock(CustomObject.class);
+        when(oldCustomObject.getValue()).thenReturn(oldJsonNode);
+        when(oldCustomObject.getContainer()).thenReturn(container);
+        when(oldCustomObject.getKey()).thenReturn(key);
+
+        assertThat(oldJsonNode.toString()).isNotEqualTo(newJsonNode.toString());
+        assertThat(CustomObjectSyncUtils.hasIdenticalValue(oldCustomObject, newCustomObjectdraft)).isTrue();
+    }
+
+    @Test
+    void hasIdenticalValue_WithDifferentNestedJsonNode_ShouldNotBeIdentical() {
+
+        JsonNode oldNestedJson = JsonNodeFactory.instance.objectNode()
+                .put("username", "Peter")
+                .put("userId", "123-456-789");
+        JsonNode newNestedJson = JsonNodeFactory.instance.objectNode()
+                .put("userId", "129-382-189")
+                .put("username", "Peter");
+
+        JsonNode oldJsonNode = JsonNodeFactory.instance.objectNode()
+                .set("nestedJson", oldNestedJson);
+
+        JsonNode newJsonNode = JsonNodeFactory.instance.objectNode()
+                .set("nestedJson", newNestedJson);
+
+
+        newCustomObjectdraft = CustomObjectDraft.ofUnversionedUpsert(container, key, newJsonNode);
+        oldCustomObject = mock(CustomObject.class);
+        when(oldCustomObject.getValue()).thenReturn(oldJsonNode);
+        when(oldCustomObject.getContainer()).thenReturn(container);
+        when(oldCustomObject.getKey()).thenReturn(key);
+
+        assertThat(oldJsonNode.toString()).isNotEqualTo(newJsonNode.toString());
+        assertThat(CustomObjectSyncUtils.hasIdenticalValue(oldCustomObject, newCustomObjectdraft)).isFalse();
+    }
+
+    @Test
+    void batchElements_WithValidSize_ShouldReturnCorrectBatches() {
+        final int numberOfCustomObjectDrafts = 160;
+        final int batchSize = 10;
+        final ArrayList<CustomObjectDraft<JsonNode>> customObjectDrafts = new ArrayList<>();
+
+        for (int i = 0; i < numberOfCustomObjectDrafts; i++) {
+            customObjectDrafts.add(CustomObjectDraft
+                    .ofUnversionedUpsert("container_" + i, "key_" + i,
+                            JsonNodeFactory.instance.objectNode().put("field", "field_" + i)));
+        }
```

Comment:
```suggestion
        final List<CustomObjectDraft<JsonNode>> field = IntStream.of(160)
                                                                 .mapToObj(i -> CustomObjectDraft
                                                                         .ofUnversionedUpsert("container_" + i, "key_" + i,
                                                                                 JsonNodeFactory.instance.objectNode().put("field", "field_" + i))
                                                                 )
                                                                 .collect(Collectors.toList());
```

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/561/commits/1e6b7b4a0e518cc8fb14c20854026e165de94e00

Reply:
After testing, It doesn't work. It created a single-element list with i=160

Reply:
What about now?
```
        final List<CustomObjectDraft<JsonNode>> customObjectDrafts = IntStream.range(0, 160)
                                                                 .mapToObj(i -> CustomObjectDraft
                                                                         .ofUnversionedUpsert("container_" + i, "key_" + i,
                                                                                 JsonNodeFactory.instance.objectNode().put("field", "field_" + i))
                                                                 )
                                                                 .collect(Collectors.toList());
```

Reply:
Thanks.
`batchElements` is removed from `CustomObjectSyncUtils`, so i think unit test cases are not needed now.

Reply:
Damn, after all the efforts 😄